lib/rspec-puppet-facts.rb

```diff
@@ -395,10 +399,15 @@ def self.facter_version_for_puppet_version(puppet_version)
   ensure
     fd.close if fd
   end
+
+  def self.legacy_facts_default_for_puppet_version(puppet_version)
+    Gem::Version.new(puppet_version) < Gem::Version.new('8.0.0')
+  end
 end
 
 RSpec.configure do |c|
   c.add_setting :default_facter_version,
     :default => RspecPuppetFacts.facter_version_for_puppet_version(Puppet.version)
   c.add_setting :facterdb_string_keys, :default => false
+  c.add_setting :legacy_facts, :default => RspecPuppetFacts.legacy_facts_default_for_puppet_version(Puppet.version)
```

Comment:
Following discussion with @ekohl perhaps this setting should get bumped up into `rspec-puppet` itself so that it can be used by both `rspec-puppet-facts` and the native facter implementation there.